var/spack/repos/builtin/packages/revbayes/package.py

```diff
@@ -29,26 +30,43 @@ class Revbayes(CMakePackage):
 
     def url_for_version(self, version):
         if version > Version('1.0.13'):
-            return 'https://github.com/revbayes/revbayes/archive/v{0}.tar.gz'.format(version)
+            return 'https://github.com/revbayes/revbayes/archive/{0}.tar.gz'.format(version)
         else:
             return 'https://github.com/revbayes/revbayes.archive/archive/v{0}.tar.gz'.format(version)
 
     @property
     def root_cmakelists_dir(self):
-        if self.spec.version > Version('1.0.13') and '+mpi' in self.spec:
-            return 'projects/cmake/build-mpi'
+        if self.spec.version > Version('1.0.13'):
+            return 'src'
         else:
             return 'projects/cmake/build'
 
+    @when('@1.1.0:')
+    def cmake_args(self):
+        args = []
+        if '+mpi' in self.spec:
+            args.extend([
+                self.define('MPI', 'ON'),
+                self.define('RB_EXEC_NAME', 'rb-mpi'),
+            ])
+        return args
+
+    @run_before('cmake')
+    def regenerate(self):
+        with working_dir(join_path('projects', 'cmake')):
+            mkdirp('build')
+            generate_version = Executable('./generate_version_number.sh')
+            generate_version()
+            dest = join_path('..', '..', 'src', 'revlanguage', 'utils')
+            install('GitVersion.cpp', dest)
+            regenerate = Executable('./regenerate.sh')
+            regenerate()
+
+    @when('@:1.1.0')
     @run_before('cmake')
```

Comment:
I don't think `@when` and @run_before` play nicely together, maybe @alalazo can comment. I think that's what is causing the flake8 tests to fail, although that could also be because we don't add the `# noqa` properly unless the `@when` is the line before the function name.

Reply:
Went ahead and switched back to a condition in the function.